nilearn/input_data/nifti_labels_masker.py

```diff
@@ -178,6 +187,60 @@ def __init__(self, labels_img, background_label=0, mask_img=None,
             raise ValueError("invalid value for 'resampling_target' "
                              "parameter: " + str(resampling_target))
 
+    def generate_report(self):
+        from nilearn.reporting.html_report import generate_report
+        return generate_report(self)
+
+    def _reporting(self):
+        """
+        Returns
+        -------
+        displays : list
+            A list of all displays to be rendered.
+
+        """
+        try:
+            from nilearn import plotting
+        except ImportError:
+            with warnings.catch_warnings():
+                mpl_unavail_msg = ('Matplotlib is not imported! '
+                                   'No reports will be generated.')
+                warnings.filterwarnings('always', message=mpl_unavail_msg)
+                warnings.warn(category=ImportWarning,
+                              message=mpl_unavail_msg)
+                return [None]
+
+        if 'labels_image' in self._reporting_data:
+            labels_image = self._reporting_data['labels_image']
+            labels_image_data = image.get_data(labels_image)
+            number_of_regions = np.sum(np.unique(labels_image_data) != 0)
+            self._report_content['description'] = (
+            'This reports shows the regions defined by the labels of the mask.')
+            self._report_content['number_of_regions'] = number_of_regions
+            
+            label_values = np.unique(labels_image_data).nonzero()[0]
+            columns = ['label value', 'region name', 'size (in #voxels)',
```

Comment:
I'd put the size in mm^3, because the # voxels depends on the arbitrary image resolution.